sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/key_client_options.hpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <azure/core/http/http.hpp>
+#include <azure/core/response.hpp>
+#include <azure/keyvault/common/keyvault_pipeline.hpp>
+
+#include "azure/keyvault/keys/key_vault_key.hpp"
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
+
+  enum class ServiceVersion
+  {
+    V7_0,
+    V7_1
+  };
+
+  struct KeyClientOptions
+  {
+    ServiceVersion Version;
+    Azure::Core::Http::RetryOptions RetryOptions;
+    Azure::Core::Http::TransportPolicyOptions TransportPolicyOptions;
+
+    KeyClientOptions(ServiceVersion version = ServiceVersion::V7_1) : Version(version) {}
+
+    std::string GetVersionString()
```

Comment:
I don't think we want this function at all. The client ctor can read the option's version and produce the right string which is an option to an api-version policy that gets added to the pipeline. We probably need to create the policy and its options struct

Reply:
I would like to make this improvement after the first min functional version